examples/usercmodule/cexample/examplemodule.c

```diff
@@ -4,6 +4,14 @@
 // Used to get the time in the Timer class example.
 #include "py/mphal.h"
 
+// Enable this if you have resources that should
+// be de-initialised after use.
+#define ENABLE_DEINIT_FUNCTION  (1)
+
+#if ENABLE_DEINIT_FUNCTION && !MICROPY_ENABLE_FINALISER
+#error example deinit requires: MICROPY_ENABLE_FINALISER
+#endif
```

Comment:
I think remove this `ENABLE_DEINIT_FUNCTION` option and unconditionally have the finaliser there.